examples/plot_calibration_curve.py

```diff
@@ -0,0 +1,134 @@
+"""
+==============================
+Probability Calibration curves
+==============================
+
+When performing classification one often wants to predict not only the class
+label, but also the associated probability. This probability gives some
+kind of confidence on the prediction. This example demonstrates how to display
+how well calibrated the predicted probabilities are and how to calibrate an
+uncalibrated classifier.
+
+The experiment is performed on an artificial dataset for binary classification
+with 100.000 samples (1.000 of them are used for model fitting) with 20
+features. Of the 20 features, only 2 are informative and 10 are redundant. The
+first figure shows the estimated probabilities obtained with logistic
+regression, Gaussian naive Bayes, and Gaussian naive Bayes with both isotonic
+calibration and sigmoid calibration. The calibration performance is evaluated
+with Brier score, reported in the legend (the smaller the better). One can
+observe here that logistic regression is well calibrated while raw Gaussian
+naive Bayes performs very badly. This is because of the redundant features
+which violate the assumption of feature-independence and result in an overly
+confident classifier, which is indicated by the typical transposed-sigmoid
+curve.
+
+Calibration of the probabilities of Gaussian naive Bayes with isotonic
+regression can fix this issue as can be seen from the nearly diagonal
+calibration curve. Sigmoid calibration also improves the brier score slightly,
+albeit not as strongly as the non-parametric isotonic regression. This can be
+attributed to the fact that we have plenty of calibration data such that the
+greater flexibility of the non-parametric model can be exploited.
```

Comment:
The Sigmoid calibration does not seem to work as expected on the inverted sigmoid shape of NB models. Is this an intrinsic limitation of the parametrization of the Platt method that expects a non-inverted sigmoid or a limitation of our own implementation? Based on my understanding of Platt's method I would think the former (that is a limitation of the parametrization of the method that is not suitable for NB models). If this is correct I think we should rewrite that paragraph to make this limitation more explicit.